src/Maker/MakeDockerDatabase.php

```diff
@@ -0,0 +1,154 @@
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+final class MakeDockerDatabase extends AbstractMaker
+{
+    private $fileManager;
+    private $composeFilePath;
+
+    /**
+     * @var ComposeFileManipulator
+     */
+    private $composeFileManipulator;
+
+    /**
+     * @var string type of database selected by the user
+     */
+    private $databaseChoice;
+
+    /**
+     * @var string Service identifier to be set in docker-compose.yaml
+     */
+    private $serviceName = 'database';
+
+    /**
+     * @var string Version set in docker-compose.yaml for the service. e.g. latest
+     */
+    private $serviceVersion = 'latest';
+
+    public function __construct(FileManager $fileManager)
+    {
+        $this->fileManager = $fileManager;
+        $this->composeFilePath = sprintf('%s/docker-compose.yaml', $this->fileManager->getRootDirectory());
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:docker:database';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConfig): void
+    {
+        $command
+            ->setDescription('Adds a database container to your docker-compose.yaml file.')
+        ;
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        $io->section('- Docker Compose Setup-');
+
+        $composeFileContents = '';
+        $statusMessage = 'Existing docker-compose.yaml not found: a new one will be generated!';
+
+        if ($this->fileManager->fileExists($this->composeFilePath)) {
+            $composeFileContents = $this->fileManager->getFileContents($this->composeFilePath);
+
+            $statusMessage = 'We found your existing docker-compose.yaml: Let\'s update it!';
+        }
+
+        $io->text($statusMessage);
+
+        $this->composeFileManipulator = new ComposeFileManipulator($composeFileContents);
+
+        $io->newLine();
+
+        $this->databaseChoice = strtolower($io->choice(
+            'Which database service will you be creating?',
+            ['MySQL', 'MariaDB', 'Postgres']
+        ));
+
+        $io->text([sprintf(
+            'For a list of supported versions, check out https://hub.docker.com/_/%s',
+            $this->databaseChoice
+        )]);
+
+        $this->serviceVersion = $io->ask('What version would you like to use?', DockerDatabaseServices::getSuggestedServiceVersion($this->databaseChoice));
+
+        if ($this->composeFileManipulator->serviceExists($this->serviceName)) {
+            $io->comment(sprintf('A <fg=yellow>"%s"</> service is already defined.', $this->serviceName));
+            $io->newLine();
+
+            $envTest[] = 'If you are using the Symfony Binary, it will expose the env_var associated with this service.';
```

Comment:
`env_var` => `env var` or `environment variable` - not sure `env_var` specifically is a thing :)

And maybe:

> ...it will expose the connection config of this service as an environment variable.
> The *name* of the service determines the name of the environment variable.
> For example, if you name the service 'database', the binary will expose a DATABASE_URL environment variable.

Not sure this is perfect, but it feels like it needs to get "filled out" a bit more

Reply:
refactored